msbuild/Xamarin.Mac.Tasks.Core/Tasks/DetectSdkLocations.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.MacDev.Tasks;
+using Xamarin.MacDev;
+
+namespace Xamarin.Mac.Tasks
+{
+	public class DetectSdkLocationsTaskBase : Task
+	{
+		#region Inputs
+
+		public string SessionId { get; set; }
+
+		// This is also an input
+		[Output]
+		public string SdkVersion {
+			get; set;
```

Comment:
This isn't really a move, though. The original files where turned into bases (in a new Core project). The signature of the originals is unchanged, by them inheriting the bases. The only changes in the code of the bases is to add a SessionId property and the naming. My problem with doing it in separate commits is that I will potentially have a commit that will not build, and that breaks Minimum Unit of Cherry-Picking. None of the code in the bases changed, though.